src/Middleware/ErrorCatcher.php

```diff
@@ -128,13 +128,14 @@ public function forceContentType(string $contentType): self
 
     public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
     {
-        $t = null;
         try {
             return $handler->handle($request);
         } catch (Throwable $t) {
-            $this->eventDispatcher?->dispatch(new ApplicationError($t));
-        } finally {
-            /** @psalm-suppress PossiblyNullArgument $t is set in catch() statement */
+            try {
+                $this->eventDispatcher?->dispatch(new ApplicationError($t));
+            } catch (Throwable) {
+                // ignore exceptions thrown in event handlers
```

Comment:
Has idea how check errors in event handlers?

Reply:
Wrap these two exceptions into a big one with linear history :)

```mermaid
graph TD;
A[aggregated error] --> B
B[dispatching error] --> C[another error]

```

Reply:
I'd error instead of ingoring it since it will be nightmare to debug. See https://github.com/yiisoft/yii2/blob/master/framework/base/ErrorHandler.php#L161